### ISSUE TITLE
Holding right mouse button and moving mouse defocuses app

### ISSUE THREAD

**Issue by jamest212** [moonlight-stream/moonlight-android#641]
I'm using a Note 9 on Android 8.1, and a USB Mouse/Keyboard combo connected via OTG.

Generally it works fine, except that I've found that when I hold the RMB down for a second or so and then move the cursor, it "defocuses" the mouse from the app out to the Android level (i.e., if I move to the screen edge, it will make the status bar appear).

This makes it hard to do things like use RMB hold to look around in MMOs, as the cursor will untie from the app until I click again to re-engage.

I tested out using Steam streaming and didn't see this issue (but had a worse problem - RMB seems bound to Escape in that app).

**Comment by cgutman** [comment]
I think this has to do with Samsung incorrectly reporting a right mouse press as the back button (you can see this by right clicking outside of a streaming session). Fortunately, I just got a Note 9 so I should be able to reproduce the issue

**Comment by jamest212** [comment]
I see what you mean. I was confused because it doesn't outright close the app - maybe due to other recent updates I saw on this issue? But in other apps it does behave just like the back from the navigation bar. Anyway, thank you for having a look.

**Comment by jamest212** [comment]
One other thing I found to be strange - when I first start a streaming session, the first time I right click hold and move my mouse around, the issue doesn't occur. If I release and hold again, then start moving, the issue occurs.

**Comment by cgutman** [comment]
I was not able to reproduce this issue on my Note 9. I started a stream, right clicked a few times, then held the right mouse button for a few seconds, moved the mouse to the top of the screen, and didn't see the status bar appear. Am I missing a step?

Are you using DeX?

**Comment by jamest212** [comment]
Not using DeX.

I just tried with my home wired USB KB and mouse, and was able to reproduce again. Previously I was using a wireless KB/mouse combo, though I wouldn't suspect the app or OS would differentiate anyway. Both are via USB OTG connector.

This time I just started up my "Desktop" shortcut (using the mstsc.exe trick from the setup guide), right clicked once on the desktop (no problem), then right click and hold - don't even really need to move the mouse but slightly, then I can see the OS cursor pop up signaling that the mouse is no longer engaged to the app. Status bar will show up if I move to the top of the screen.

I've tried different settings on the navigation bar (Settings -> Display -> Navigation Bar) - not sure how you have yours set. Tried turning off the lock button (so it always shows - but it still hides during Moonlight streaming for some reason?), or turning on the lock button, locking it in hidden with the lock button, etc. Doesn't seem to be affected by the navigation bar settings. The real issue is the mouse disengaging from the app - the navigation bar just may or may not show as a side effect of that.

**Comment by jamest212** [comment]
Also, in another strange bit, I found that if I did "something" with the keyboard while I was holding down I could avoid the problem from happening. In my case, I was holding down W to walk forward in game and hit A or D a couple times to strafe, all while holding down the mouse key, and this would for some reason avoid the problem. If I just held down W, it would not. It was as if it took the multiple key presses to interfere and avoid the issue.

**Comment by jamest212** [comment]
Well, FWIW I tried remotr for the first time and managed to recreate the same problem.

So it's something on my side, not sure what, if you can't reproduce, but not a moonlight issue, I guess.

**Comment by jamest212** [comment]
Ok, last post from me unless you think I can give you more info.

Happens across many different apps so seems to be how Samsung treats the RMB as the software back button, I guess, like you said.

Remotr has exactly the same issue as Moonlight, and SteamLink (standard version) seems to treat it like an Escape press in some way. I was previously using only USB OTG, but now I've also tried with a Bluetooth mouse, and that also makes no difference.

What ended up being my fix is to use the Samsung App Store version of SteamLink. I guess unsurprisingly they've conquered their own janking of the phone/kernel and they recently added mouse/keyboard support. Issues with right click and hold do not seem to occur at all (yet). I would still prefer Moonlight since the streaming performance is usually better, lighter-weight and less prone to screwups than the Steam-based ones (IME).

I also investigated whether it could be due to the Galaxy SteamLink having the Samsung game center recognition and whether that might change the behavior - but no dice. Enabled Moonlight in the game center and then blocked EVERYTHING, like edge panels and so on, but the issue still occurred.

So in short, seems like there is a software workaround to the issue, but I know nothing about these things so no idea if it's something a third party developer has the ability to do.

**Comment by jamest212** [comment]
There was a new GFE release today so I tried again. Problem still technically occurs, but I seem to have accidentally found a workaround that might help give you a clue as to how to fix it on the software side:

I fired up a desktop stream like normal, right clicked a couple times, then held down and got the "defocusing" issue again. I am trying this on my home computer, so I have the mouse and keyboard hooked to a button-toggle USB hub. I briefly toggled over to my desktop, then toggled control back over to the Note 9 and then was unable to reproduce the issue.

Without toggling the USB hub, I closed the stream (on the phone side only) and then re-entered the session, and again had the defocusing issue on holding RMB. Toggled back and forth, and then couldn't make it happen again.

Not sure what the minimal conditions for this - is it just losing the mouse connection then regaining it without re-entering the session? I will try with my bluetooth mouse tomorrow to see if just turning that off and on during the session works similarly.

I don't think this is related to the GFE update, but obviously can't be 100% sure. Nothing in the patch notes to suggest it is. I just didn't try this specific sequence of actions before because... why would that work?

Hope this helps.

**Comment by jamest212** [comment]
So turning the mouse off and on did not work, but totally disabling bluetooth on the phone and then turning it back on did.

**Comment by cgutman** [comment]
Hopefully this is fixed in the Pie update.

**Comment by ZMasterFurry** [comment]
Just to add, I'm having this problem currently (new to Moonlight) and I am on a Note 9 as well.  My problem seems to be the same as jamest212, he describes the problem accurately,  and currently with the knowledge I have, I don't have a  fix. Just stating the problem is still around.
